opencti-platform/opencti-front/src/private/components/data/connectors/parseWorkErrors.tsx

```diff
@@ -103,76 +104,115 @@ export interface PartialParsedWorkMessage {
 
 export type ParsedWorkMessage = FullParsedWorkMessage | PartialParsedWorkMessage;
 
-const criticalErrorTypes = [
+const criticalDocCodes = [
+  // Doc_code
+  'ELEMENT_ID_COLLISION',
+  // Name
   'MULTIPLE_REFERENCES_ERROR',
   'UNSUPPORTED_ERROR',
   'DATABASE_ERROR',
   'INTERNAL_SERVER_ERROR',
 ];
 
-const warningErrorTypes = [
+const warningDocCodes = [
+  // Doc_code
+  'INCORRECT_INDICATOR_FORMAT',
+  'INCORRECT_OBSERVABLE_FORMAT',
+  'RESTRICTED_ELEMENT',
+  'MULTIPLE_REFERENCES_FOUND',
+  'SELF_REFERENCING_RELATION',
+  'INSUFFICIENT_CONFIDENCE_LEVEL',
+  'ELEMENT_NOT_FOUND',
+  // Name
   'VALIDATION_ERROR',
   'MULTIPLE_ENTITIES_ERROR',
   'ACL_ERROR',
   'MISSING_REFERENCE_ERROR',
 ];
 
+export const getLevel = (code: string): ErrorLevel => {
+  if (criticalDocCodes.includes(code)) return 'Critical';
+  if (warningDocCodes.includes(code)) return 'Warning';
+  return 'Unclassified';
+};
+
+export const parseError = (error: NonNullable<NonNullable<WorkMessages>[number]>): ParsedWorkMessage => {
+  // Try/Catch to prevent JSON.parse Exception
+  try {
+    const parsedSource = JSON5.parse(error.source ?? '');
+    const source = parsedSource.type === 'bundle' ? parsedSource.objects[0] : parsedSource;
+    const message = JSON5.parse((error.message ?? ''));
+    const entityId = source.id;
+    const fromId = source.source_ref;
+    const toId = source.target_ref;
+
+    const parsedError = {
+      category: message.name,
+      doc_code: message.doc_code ?? message.name,
```

Comment:
Oh I see here there is a fallback on the name. Forget about my comment, I'll resolve it now :)